### ISSUE TITLE
"Internal error" while logging non-string messages

### ISSUE THREAD

**Issue by Delgan** [kornicameister/loguru-mypy#49]
Hi! :)

It seems that some expressions are not recognized and generate an internal error in the plugin. It happens due to the `assert` here: https://github.com/kornicameister/loguru-mypy/blob/74d4af6331f20f2e4a727c911420526d813cc020/loguru_mypy/__init__.py#L57

Actually, the argument does not necessarily needs to be string. Here are some examples that generate an error.

```python
from loguru import logger

logger.info(123)
```

```python
from loguru import logger

logger.info("foo" + "bar")
```

```python
from loguru import logger

foo = "bar"
logger.info(foo)
```

```python
from loguru import logger

logger.info(f"{foobar}")
```

Which results in:

```
a.py:3: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.790
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3766, in accept
  File "mypy/checkexpr.py", line 263, in visit_call_expr
  File "mypy/checkexpr.py", line 340, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 817, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 880, in check_call
  File "mypy/checkexpr.py", line 1532, in check_overload_call
  File "mypy/checkexpr.py", line 1675, in infer_overload_return_type
  File "mypy/checkexpr.py", line 876, in check_call
  File "mypy/checkexpr.py", line 988, in check_callable_call
  File "mypy/checkexpr.py", line 725, in apply_function_plugin
  File "/home/delgan/Programmation/loguru-mypy/loguru_mypy/__init__.py", line 99, in _loguru_logger_call_handler
    assert isinstance(log_msg_expr, StrExpr), type(log_msg_expr)
AssertionError: <class 'mypy.nodes.IntExpr'>
a.py:3: : note: use --pdb to drop into pdb
```

**Comment by ThibaultLemaire** [comment]
As mentioned in https://github.com/kornicameister/loguru-mypy/issues/42#issuecomment-753906878 I'd like to work the 3rd case (variables):

```python
from loguru import logger

foo = "bar"
logger.info(foo)
```

**Comment by kornicameister** [comment]
Ok, I believe we have number 3 fixed, like @ThibaultLemaire did in b608dc03a7ebb9d71f9c8a7052a931c48467838e. 
I am closing this issue but fear not ( :P ) there will be issues scoped to figure out remaining points brought up by @Delgan.

I believe those are:
- `OpExpr` (an expression like `2+1`).
- `CallExpr` (which seems to be a case for **f-string** but only those that use an argument)

**Comment by kornicameister** [comment]
Looks like another funny bit has been implemented as part of #81. That will give us a single argument variation on non-str arguments, if i am correct.
